3-extra/2-array-of-objects.js

```diff
@@ -10,73 +10,97 @@
     Each title in the resulting array should be the highest rated book in its genre.
 */
 
-function getHighestRatedInEachGenre(books) {
-    // TODO
-}
-
-
+// function getHighestRatedInEachGenre(books) {
+//     let allBooks = books.map((books) => Object.assign(books));
+//     return allBooks;
+// }
+// console.log(getHighestRatedInEachGenre(BOOKS));
 /* ======= Book data - DO NOT MODIFY ===== */
-const BOOKS = [
-    {
+const BOOKS = [{
         title: "The Lion, the Witch and the Wardrobe",
         genre: "children",
-        rating: 4.7
+        rating: 4.7,
     },
     {
         title: "Sapiens: A Brief History of Humankind",
         genre: "non-fiction",
-        rating: 4.7
+        rating: 4.7,
     },
     {
         title: "Nadiya's Fast Flavours",
         genre: "cooking",
-        rating: 4.7
+        rating: 4.7,
     },
     {
         title: "Harry Potter and the Philosopher's Stone",
         genre: "children",
-        rating: 4.8
+        rating: 4.8,
     },
     {
         title: "A Life on Our Planet",
         genre: "non-fiction",
-        rating: 4.8
+        rating: 4.8,
     },
     {
         title: "Dishoom: The first ever cookbook from the much-loved Indian restaurant",
         genre: "cooking",
-        rating: 4.85
+        rating: 4.85,
     },
     {
         title: "Gangsta Granny Strikes Again!",
         genre: "children",
-        rating: 4.9
+        rating: 4.9,
     },
     {
         title: "Diary of a Wimpy Kid",
         genre: "children",
-        rating: 4.6
+        rating: 4.6,
     },
     {
         title: "BOSH!: Simple recipes. Unbelievable results. All plants.",
         genre: "cooking",
-        rating: 4.6
+        rating: 4.6,
     },
     {
         title: "The Book Your Dog Wishes You Would Read",
         genre: "non-fiction",
-        rating: 4.85
+        rating: 4.85,
     },
-]
+];
 
+function category(books, genre) {
+    let children = [];
+    books.forEach((element) => {
+        if (element.genre === genre) {
+            children.push(element);
+        }
+    });
+    return children;
+}
+
+function operation(books, genre) {
+    let children = category(books, genre);
+    let result = children.sort((a, b) => b.rating - a.rating)[0];
+    return result.title;
+}
+
+function getHighestRatedInEachGenre(books) {
+    let children = operation(books, "children");
+    let non_fiction = operation(books, "non-fiction");
+    let cooking = operation(books, "cooking");
+    const result = [children, non_fiction, cooking];
+    return result;
+}
```

Comment:
Your functions could have more descriptive names. Also, think about how you might get the list of genres from the array of objects rather than having to hard code it. Don't feel you have to redo this - it's a difficult, stretch exercise and your method is understandable and works well - it's just something to think about.